### PR TITLE
Refactor/cuda/low lat pbs scratch

### PR DESCRIPTION
**Pull request by agnesLeroy**
### Checklist 

(Use '[x]' to check the checkboxes, or submit the PR and then click the checkboxes)

* [ ] Tests for the changes have been added (for bug fixes / features)
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] The PR description links to the related issue (to link an issue, use '#XXX'.)
* [ ] The tests on AWS have been launched and are successful (comment with @slab-ci cpu_test and/or @slab-ci gpu_test to trigger the tests)
* [ ] The draft release description has been updated
* [ ] Check for breaking changes (including serialization changes) and add them to commit message following the conventional commit [specification][conventional-breaking]

[conventional-breaking]: https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with-description-and-breaking-change-footer

### CODE REVIEW

**Comment by agnesLeroy** [comment]
@slab-ci gpu_test

**Comment by agnesLeroy** [comment]
@slab-ci gpu_test

**Comment by agnesLeroy** [comment]
@slab-ci gpu_test

**Comment by agnesLeroy** [comment]
@slab-ci gpu_test

**Comment by agnesLeroy** [comment]
@slab-ci gpu_test

**Comment by agnesLeroy** [comment]
@slab-ci gpu_test

**Comment by agnesLeroy** [comment]
@slab-ci gpu_test

File: concrete-cuda/cuda/src/bit_extraction.cuh [hunk 1]
```diff
@@ -127,6 +127,53 @@ __global__ void fill_lut_body_for_current_bit(Torus *lut, Torus value) {
...
 
+template <typename Torus>
+__host__ __device__ int
+get_buffer_size_extract_bits(uint32_t glwe_dimension, uint32_t lwe_dimension,
+                             uint32_t polynomial_size,
+                             uint32_t number_of_inputs) {
+
+  return sizeof(Torus) * number_of_inputs // lut_vector_indexes
+         + ((glwe_dimension + 1) * polynomial_size) * sizeof(Torus) // lut_pbs
+         + (polynomial_size + 1) * sizeof(Torus)  // lwe_array_in_buffer
```
Comment (pdroalves) [inline concrete-cuda/cuda/src/bit_extraction.cuh:138]: Are you sure the size of the input LWE is `polynomial_size + 1` and not `glwe_dimension * polynomial_size +1`?

File: concrete-cuda/src/cuda_bind.rs [hunk 2]
```diff
@@ -222,6 +222,36 @@ extern "C" {
...
+        glwe_dimension: u32,
+        polynomial_size: u32,
+        level_count: u32,
+        input_lwe_ciphertext_count: u32,
+        max_shared_memory: u32,
+        allocate_gpu_memory: bool,
+    );
+
+    /// This scratch function allocates the necessary amount of data on the GPU for
+    /// the amortized PBS on 64 bits inputs, into `pbs_buffer`. It also configures SM
```
Comment (pdroalves) [inline concrete-cuda/src/cuda_bind.rs:241]: amortized => low_latency

File: concrete-cuda/src/cuda_bind.rs [hunk 3]
```diff
@@ -222,6 +222,36 @@ extern "C" {
         pbs_buffer: *mut *mut i8,
     );
 
+    /// This scratch function allocates the necessary amount of data on the GPU for
+    /// the amortized PBS on 32 bits inputs, into `pbs_buffer`. It also configures SM
```
Comment (pdroalves) [inline concrete-cuda/src/cuda_bind.rs:226]: amortized => low_latency

**Review by pdroalves — CHANGES_REQUESTED** [review]
Nice job!
I just found some minor problem in two comments and left a question regarding some lwe array sizes.

File: concrete-cuda/cuda/src/bit_extraction.cuh (hunk 1 above)
Comment (agnesLeroy) [reply concrete-cuda/cuda/src/bit_extraction.cuh:138]: I have no idea... :disappointed:

Comment (agnesLeroy) [reply concrete-cuda/cuda/src/bit_extraction.cuh:138]: It should be glwe_dimension * polynomial_size + 1 indeed.

Comment (agnesLeroy) [reply concrete-cuda/cuda/src/bit_extraction.cuh:138]: I'll fix it

**Comment by agnesLeroy** [comment]
@slab-ci gpu_test

**Review by pdroalves — APPROVED** [review]

**Pull request merged**